### ISSUE TITLE
Tests in ValueTypeTests skipped in Valhalla builds

### ISSUE THREAD

**Issue by hangshao0** [eclipse-openj9/openj9#16284]
Found by @a7ehuo in Valhalla nightly build: https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdkValhalla_j9_sanity.functional_x86-64_linux_valhalla_Nightly_testList_0/673

Only tests that have priority=1 in `ValueTypeTests.java` get executed. All other tests with priority>1 are skipped unexpectedly.

**Comment by a7ehuo** [comment]
Valhalla functional tests are pretty much the only functional  tests that explicitly set priority. @llxia I can reproduce the similar issue with non-Valhalla test after adding priority. I experimented with `jsr335InterfacePrivateMethod`. I added `priority=1`, `priority=2`, `priority=3` to the tests in `Test_ITestRunner.java` and `priority=1` to the tests in `Test_ValidateReflectionWithExtensionMethods.java`. None of the tests from `Test_ITestRunner.java` and `Test_ValidateReflectionWithExtensionMethods.java` get invoked. Could this be a test framework issue? 

Attach the modified test files and logs files here: [functional_test_skip_tests_with_priority.zip](https://github.com/eclipse-openj9/openj9/files/9975198/functional_test_skip_tests_with_priority.zip)


```
	<test name="jsr335InterfacePrivateMethod">
		<classes>
			<class name="org.openj9.test.jsr335.interfacePrivateMethod.Test_ITestRunner" />
			<class name="org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles" />
			<class name="org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods" />
		</classes>
	</test>
```

**Comment by llxia** [comment]
I can reproduce the issue in TestNG 6.14.2. The issue seems resolved in TestNG 7.6.1. I ran the above case and got 66 tests.

```
...
... TestNG 7.6.1 by Cédric Beust (cedric@beust.com)
...

PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_ITest_private_non_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ITestRunner.testRunTest_private_static_method_with_InvokeStatic
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetMethodsOnF
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflect_child_public_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_ITest_public_non_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_static_child_public_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_static_ancestor_public_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflectSpecial_interface_ITest_private_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetMethodsOnK
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_child_public_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_virtual_child_public_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_child_public_non_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_special_ancestor_public_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetDeclaredMethodsOnE
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ITestRunner.testRunTest_private_non_static_method_with_InvokeVirtual
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetDeclaredMethodsOnK
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetDeclaredMethodsOnJ
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ITestRunner.testRunTest_private_static_method_with_InvokeVirtual
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetMethodsOnJ
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.testFindMethodsUsingGetMethods
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetDeclaredMethodsOnD
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ITestRunner.testRunTest_private_non_static_method_with_InvokeSpecial
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ITestRunner.testRunTest_private_non_static_method_with_InvokeInterface
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetMethodsOnC
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetMethodsOnG
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetDeclaredMethodsOnF
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_child_private_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_virtual_interface_ITest_private_non_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_ITest_public_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_ITest_private_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_virtual_child_public_non_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_special_ancestor_private_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_virtual_ancestor_public_non_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_special_child_private_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.testFindMethodsUsingGetDeclaredMethods
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_special_child_clinit
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_special_child_private_non_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ITestRunner.testRunTest_private_static_method_with_InvokeInterface
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetDeclaredMethodsOnC
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_findVirtual_interface_ITest_private_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflectSpecial_interface_ITest_private_non_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetDeclaredMethodsOnH
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_static_child_private_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_static_ancestor_private_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetDeclaredMethodsOnG
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflect_child_private_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_child_private_non_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetMethodsOnE
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_special_child_public_non_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflectSpecial_child_private_non_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflect_interface_ITest_private_non_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ITestRunner.testRunTest_private_non_static_method_with_InvokeStatic
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetMethodsOnD
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ITestRunner.testRunTest_private_static_method_with_InvokeSpecial
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_special_ancestor_private_non_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_virtual_child_private_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ValidateReflectionWithExtensionMethods.testGetMethodsOnH
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflectSpecial_child_private_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_special_child_public_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflectSpecial_child_public_non_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflect_child_public_non_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflectSpecial_child_public_static_method
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_special_ancestor_public_non_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_static_child_private_non_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_find_static_child_public_non_static
PASSED: org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflect_interface_ITest_private_static_method

===============================================
    jsr335InterfacePrivateMethod
    Tests run: 66, Failures: 0, Skips: 0
===============================================


===============================================
GeneralTest suite
Total tests run: 66, Passes: 66, Failures: 0, Skips: 0
===============================================


jsr335_interfacePrivateMethod_0_PASSED
```

**Comment by llxia** [comment]
Created an issue to [upgrade TestNG to v7.6.1](https://github.com/adoptium/aqa-tests/issues/4141)
